src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -196,87 +200,78 @@ export default class AffirmPaymentStrategy implements PaymentStrategy {
         return shippingInformation;
     }
 
-    private _getItems(): AffirmItem[] {
-        const state = this._store.getState();
-        const cart = state.cart.getCart();
-
-        if (!cart) {
-            throw new MissingDataError(MissingDataErrorType.MissingCart);
-        }
+    private _getItems(amountTransformer: AmountTransformer, order: Order): AffirmItem[] {
         const items: AffirmItem[] = [];
 
-        for (const item of cart.lineItems.physicalItems) {
+        for (const item of order.lineItems.physicalItems) {
             items.push({
                 display_name: item.name,
                 sku: item.sku,
-                unit_price: item.salePrice,
+                unit_price: amountTransformer.toInteger(item.salePrice),
                 qty: item.quantity,
                 item_image_url: item.imageUrl,
                 item_url: item.url,
             });
         }
 
-        for (const item of cart.lineItems.digitalItems) {
+        for (const item of order.lineItems.digitalItems) {
             items.push({
                 display_name: item.name,
                 sku: item.sku,
-                unit_price: item.salePrice,
+                unit_price: amountTransformer.toInteger(item.salePrice),
                 qty: item.quantity,
                 item_image_url: item.imageUrl,
                 item_url: item.url,
             });
         }
 
-        if (cart.lineItems.customItems) {
-            for (const item of cart.lineItems.customItems) {
+        for (const item of order.lineItems.giftCertificates) {
+            items.push({
+                display_name: item.name,
+                sku: '',
+                unit_price: amountTransformer.toInteger(item.amount),
+                qty: 1,
+                item_image_url: '',
+                item_url: '',
+            });
+        }
+
+        if (order.lineItems.customItems) {
+            for (const item of order.lineItems.customItems) {
                 items.push({
                     display_name: item.name,
                     sku: item.sku,
-                    unit_price: item.listPrice,
+                    unit_price: amountTransformer.toInteger(item.listPrice),
```

Comment:
What´s the difference between `item.listPrice` and `item.SalePrice` and why for some item types we use one and for other types we use the other?

Reply:
for custom items only listPrice is available, sale prices may include taxes and discounts into consideration